core/src/main/scala/kafka/server/LogDirFailureChannel.scala

```diff
@@ -38,7 +41,8 @@ class LogDirFailureChannel(logDirNum: Int) {
    * If the given logDir is not already offline, add it to the
    * set of offline log dirs and enqueue it to the logDirFailureEvent queue
    */
-  def maybeAddLogFailureEvent(logDir: String): Unit = {
+  def maybeAddLogFailureEvent(logDir: String, msg: => String, t: IOException): Unit = {
```

Comment:
Not related to this PR, but it would probably be better if `logDir` was a `Path` (or `File`, but the former is the recommended once since Java 7). Generally, it's better to use types other than `String`. Same for `takeNextLogFailureEvent`.